### PR TITLE
Replace usages of `_source.mode` in documentation

### PR DESCRIPTION
**Pull request by salvatore-campagna**
We will deprecate the `_source.mode` mapping level configuration
in favor of the index-level `index.mapping.source.mode` setting.
As a result, we go through the documentation and update it to reflect
the introduction of the setting.

### CODE REVIEW

File: docs/reference/mapping/types/flattened.asciidoc [hunk 1]
```diff
@@ -407,8 +423,16 @@ For example:
...
+    "index": {
+      "mapping": {
+        "source": {
+          "mode": "synthetic"
+        }
+      }
+    }
+  },
   "mappings": {
-    "_source": { "mode": "synthetic" },
```
Comment (martijnvg) [inline docs/reference/mapping/types/flattened.asciidoc:411]: I don't fully understand why this failures happens for flattened field doc tests: https://buildkite.com/elastic/elasticsearch-pull-request/builds/36289#01928be4-ffbb-4def-8006-256e92e1e5d5/6-13640

`foo` is returned in an extra array. Strange that this failure happens with a doc change.

**Review by martijnvg — APPROVED** [review]
Doc wise LGTM 👍 Left one question about the test failure.

**Comment by leemthompo** [comment]
docscheck error:

```
 1> [2024-10-14T13:49:30,191][ERROR][o.e.s.DocsClientYamlTestSuiteIT] [test] This failing test was generated by documentation starting at reference/mapping/types/date_nanos.asciidoc:synthetic-source-date-nanos-example. It may include many snippets. See docs/README.asciidoc for an explanation of test generation.
2024-10-14 18:58:01 CEST	  2> REPRODUCE WITH: ./gradlew ":docs:yamlRestTest" --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/mapping/types/date_nanos/synthetic-source-date-nanos-example}" -Dtests.seed=F66B1F7F3663ED29 -Dtests.locale=en-SE -Dtests.timezone=America/Punta_Arenas -Druntime.java=22
2024-10-14 18:58:01 CEST	  2> java.lang.AssertionError: Failure at [reference/mapping/types/date_nanos:156]: Expected a map containing
2024-10-14 18:58:01 CEST	    _source: a map containing
2024-10-14 18:58:01 CEST	         date: a list containing
2024-10-14 18:58:01 CEST	              0: expected "2014-01-01T12:10:30.000Z" but was "2015-01-01T12:10:30.000Z"
2024-10-14 18:58:01 CEST	              1: expected "2015-01-01T12:10:30.000Z" but was "2014-01-01T12:10:30.000Z"
```

**Review by lkts — APPROVED** [review]

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Comment by salvatore-campagna** [comment]
@lkts I left out `binary.asciidoc` because that uses

```
public static final TypeParser PARSER = new TypeParser(
        (n, c) -> new Builder(n, c.getIndexSettings().getMode().isSyntheticSourceEnabled())
    );
```

which ignores the setting.

**Comment by salvatore-campagna** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/114917)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**